scripts/swap-beacon/src/index.ts

```diff
@@ -246,12 +249,25 @@ export class SwapBeacon {
     );
   }
 
+  heartbeat() {
+    if (this.pingTimeout !== undefined) clearTimeout(this.pingTimeout);
+
+    this.pingTimeout = setTimeout(() => {
+      console.error("Received no ping. Terminating connection.");
+      this.client.websocket.terminate();
```

Comment:
instead of just terminating in this case, what about retrying to establish the connection, at least once or twice? if you have multiple ping timeouts in a session, you can just terminate the websocket and end the session?

Reply:
When the process exits, k8s will just spawn a new one which is effectively retrying. The service is pretty stateless and a complete restart seems fine but we can add retry mechanisms if it's costly to do a full container restart.